src/neg1c.jl

```diff
@@ -0,0 +1,143 @@
+        α[n] = gamma((2*n+1)/BigInt(2))*gamma(1+n)*HypergeometricFunctions._₂F₁general2((1+n)/BigInt(2),(2+n)/BigInt(2),(2*n+3)/BigInt(2),1/t^2)/(t*2*gamma(n)*gamma((2*n+3)/BigInt(2))*HypergeometricFunctions._₂F₁general2(n/BigInt(2),(n+1)/BigInt(2),(2*n+1)/BigInt(2),1/t^2))
+    end
+end
+
+# takes a previously computed vector of α_{n,n-1}(t) that has been increased in size and fills in the missing data guided by indices in inds via forward recurrence
+function αcoefficients!(α, t, inds)
+    @inbounds for n in inds
+       α[n] = (t*(2*n-1)/n-(n-1)/(n*α[n-1]))
+    end
+end
+
+# takes a previously computed vector of α_{n,n-1}(t) that has been increased in size and fills in the missing data guided by indices in inds, using a final condition computation and backward recurrence
+function backαcoeff!(α, t, inds)
+    α[end] = αdirect(BigInt(length(α)),t)
+    @inbounds for n in reverse(inds)[1:end-1]
+       α[n-1] = BigInt(n-1)/(t*(2*n-1)-n*α[n]) #BigInt(1-n)/(t-2*BigInt(n)*t+BigInt(n)*α[n])
+    end
+end
+
+# cached implementation using stable back recurrence to fill data
+mutable struct neg1c_αcfs{T} <: AbstractCachedVector{T}
+    t::T
+    data::Vector{T}
+    datasize::Tuple{Int}
+    array
+    neg1c_αcfs{T}(t::T) where T = new{T}(t, [initialα(2*t-1),αdirect(2,2*t-1)], (2,))
+end
+neg1c_αcfs(t::T) where T = neg1c_αcfs{T}(t)
+size(α::neg1c_αcfs) = (ℵ₀,)
+cache_filldata!(α::neg1c_αcfs, inds) = backαcoeff!(α.data, 2*α.t-1, inds)
+
+function resizedata!(α::neg1c_αcfs, nm) 
+    olddata = copy(α.data)
+    νμ = length(olddata)
+    nm = maximum(nm)
+    nm = max(νμ,nm)
+    if νμ ≠ nm
+        α.data = similar(olddata,maximum(nm))
+        α.data[1:νμ] = olddata[1:νμ]
+    end
+    if maximum(nm) > νμ
+        inds = Array(νμ-1:maximum(nm))
+        cache_filldata!(α, inds)
+        α.datasize = (nm,)
+    end
+    α
+end
+
+###
+#   cached implementation of normalization constants
+###
+mutable struct neg1c_normconstant{T} <: AbstractCachedVector{T}
+    t::T
+    data::Vector{T}
+    datasize::Tuple{Int}
+    array
+    neg1c_normconstant{T}(t::T) where T = new{T}(t, neg1c_normconstinitial(t,10), (10,))
+end
+neg1c_normconstant(t::T) where T = neg1c_normconstant{T}(t)
+size(B::neg1c_normconstant) = (ℵ₀,)
+cache_filldata!(B::neg1c_normconstant, inds) = neg1c_normconstextension!(B.data, inds, B.t)
+
+function resizedata!(B::neg1c_normconstant, nm) 
+    olddata = copy(B.data)
+    νμ = length(olddata)
+    nm = maximum(nm)
+    nm = max(νμ,nm)
+    if νμ ≠ nm
+        B.data = similar(olddata,maximum(nm))
+        B.data[1:νμ] = olddata[1:νμ]
+    end
+    if maximum(nm) > νμ
+        inds = Array(νμ-1:maximum(nm))
+        cache_filldata!(B, inds)
+        B.datasize = (nm,)
+    end
+    B
+end
+
+function neg1c_normconstinitial(t, N)
+    # generate α coefficients for OPs via recurrence
+    α = zeros(BigFloat,N-1)
+    α[1] = initialα(2*t-1)
+    backαcoeff!(α,2*t-1,BigInt.(2:N-1))
+    # normalization constants
+    B = [BigFloat("1")]
```

Comment:
I'm confused why `BigFloat` is hard-coded here, and will break things like autodiff. Also, `one(BigFloat)` is better as it avoids constructing a string.

Reply:
Yes, the arbitrary precision typing you are referring to in various places still needs to be cleaned up. It was all going to be agnostic about the type provided for Q.t but that was interrupted by the issue in https://github.com/JuliaApproximation/ClassicalOrthogonalPolynomials.jl/issues/28, which makes BigFloat inputs not work in the first place. The reason I originally was thinking about hardcoding BigFloat was the instability suspected in the alpha coefficients but it's actually fairly stable as implemented now, so this should not be needed by default.